### ISSUE TITLE
Add Configurability of Model Hyperparams

### ISSUE THREAD

**Issue by jkelling** [Photon-AI-Research/InSituML#190]
Use [jsonargparse](https://jsonargparse.readthedocs.io/en/latest/) for configurable params.

Configurable properties of the model should include:
* Number of ResNet blocks (between down-samplings), channels, (kernel sizes)
* Number of downsampling steps (determining latent size/compression)
* MLP hidden layer size, latent size
* ...

Training hyper params should include
* loss weights
* learning rates (potentially different learning rates for different modules, e.g. encoder+decoder, surrogate, discrimininator)

Start with a small set, discuss extensibility and extent as required.
